### ISSUE TITLE
Tailable cursor closing when reaching end of collection

### ISSUE THREAD

**Issue by michaelorr92** [Automattic/mongoose#3109]
I am attempting to stream some data from a MongoDB capped collection as follows:

```
var query = {};
var options = {tailable: true, awaitdata: true, numberOfRetries: Number.MAX_VALUE};

var stream = myColl.find(query, options).stream();

stream.on('data', function(doc){
    console.log(doc);
}).on('error', function (error){
    console.log(error);
}).on('close', function () {
    console.log('closed');
});
```

This prints the documents as expected but upon reaching the end of the collection the stream closes. Is it possible to prevent this from happening? I am looking to have the program print the data as it arrives in the collection, waiting at the end of the collection indefinitely for more data.

**Comment by vkarpov15** [comment]
This is another instance of #2922. Long story short, you need to re-open the stream because of changes to the mongodb node driver:

```
var lastTimestamp = moment.utc().subtract(60, 'minute').valueOf();

fn();

function fn(){

  var opts = {
    timestamp: { $gt: lastTimestamp },
  }

  var stream = Collection.find(opts).tailable().stream();

  stream.on('data', function(doc){
    lastTimestamp = doc.timestamp; // date field
    console.log(data);
  });

  stream.on('error', function(val) {
    console.error(val);
    stream.destroy();
    fn();
  });

}
```

**Comment by varunjayaraman** [comment]
Weird, this actually works for me on node v6.9.2, mongodb 3.4, mongoose 4.7.7, unless I'm not replicating the problem correctly

```javascript
const mongoose = require('mongoose');
const co = require('co');
mongoose.Promise = global.Promise;

// Constants
const GITHUB_ISSUE = `gh-3109`;
const BYTE_CAP = 10000000000;
const NUM_JUICES = 500;

// Globals
let Juice;

// run program
exec()
  .catch(error => {
    console.error(`Error: ${error}\n${error.stack}`);
    process.exit(2);
  });


function exec() {
  return createDb()
    .then(() => createCollection())
    .then(() => initialData())
    .then(() => startTailableCursor())
    .then(() => {
      setInterval(createEntries, 3000);
    })
}

// used for setInterval so we can create a fake stream of data
function createEntries() {
  return co(function* () {

    for (let i = 0; i < NUM_JUICES; i++) {
      yield Juice.create({ name: `juice${i}` })
    }

  });
}

// starts the cursor and sets listeners
function startTailableCursor() {
  return new Promise(function(resolve, reject) {
    const cursor = Juice
      .find()
      .tailable()
      .stream();

    cursor.on('data', (doc) => {
      console.log('doc', doc);
    });
    cursor.on('close', function() {
      console.log('closing...');
      resolve();
    });

    cursor.on('error', error => {
      console.error(error);
      cursor.destroy();
      reject(error);
    });

    return co(function* () {
      for (let i = 0; i < NUM_JUICES; i++) {
        yield Juice.create({ name: `juice${i}` })
      }
      resolve();
    }).catch(error => {
      console.error('co error', error);
      cursor.destroy();
      reject(error);
    })
  });
}

// creates initial entry in table 
function initialData() {
  return co(function* () {
    yield Juice.create({ name: 'initial-juice' });
  });
}

// creates Juice collection
function createCollection() {
  return co(function* () {
    const juiceSchema = new mongoose.Schema({
      name: String
    }, { capped: BYTE_CAP });

    Juice = mongoose.model('Juice', juiceSchema);
  });
}

function createDb() {
  return co(function* () {
    const db = mongoose.connect(`mongodb://localhost:27017/${GITHUB_ISSUE}`);
    return db;
  });
}

```

The stream waits for data and logs it as it comes in, without closing once the end of each stream of data from the `setInterval` call to `createEntries()`

**Comment by vkarpov15** [comment]
Looks like this was fixed by mongoose 4.4.8 re: https://github.com/christkv/mongodb-core/blob/2.0/HISTORY.md#133-2016-03-08, https://github.com/Automattic/mongoose/blob/master/History.md#448--2016-03-18 . Please upgrade to mongoose >= 4.4.8 for support for this.
